scalapb-runtime/jvm/src/main/scala/scalapb/StructUtils.scala

```diff
@@ -0,0 +1,99 @@
+package scalapb
+
+import java.nio.charset.Charset
+
+import com.google.protobuf.ByteString
+import com.google.protobuf.struct.{ListValue, Struct, Value}
+import com.google.protobuf.struct.Value.Kind
+import scalapb.descriptors.{FieldDescriptor, PBoolean, PByteString, PDouble, PEmpty, PEnum, PFloat, PInt, PLong, PMessage, PRepeated, PString, PValue, ScalaType}
+
+import scala.util.{Failure, Success, Try}
+
+object StructUtils {
+  case class StructDeserError(index: Int, error: String)
+  def toStruct(generatedMessage: GeneratedMessage): Struct =
+  //TODO(@thesamet)- Can we ignore the case where generatedMessage is null?
+    toStruct(generatedMessage.toPMessage.value)
+
+  def fromStruct[T <: GeneratedMessage](
+                                         companion: GeneratedMessageCompanion[T],
+                                         struct: Struct
+                                       ): Either[StructDeserError, T] = {
+    //TODO(@thesamet)- do we want to fail (return Either#Left) if a structField arrives that we don't know about? I assume we don't want to fail on that for forward compatibility but want to double check
+    val fieldDescriptorToPValue = structMapToFDMap(companion, struct.fields)
+    fieldDescriptorToPValue.right.map(PMessage).map(companion.messageReads.read)
+  }
+
+  private def structMapToFDMap(companion: GeneratedMessageCompanion[_], structFields: Map[String, Value]): Either[StructDeserError, Map[FieldDescriptor, PValue]] = {
```

Comment:
For consistency with other code this should return `PMessage`. Looks like in all cases it's followed by wrapping the map in PMessage anyway.